### PR TITLE
Implement foreign-key-aware cascaded delete

### PR DESCRIPTION
**Pull request by EamonNerbonne**
This isn't going to be a brilliant bit of code, but it's a first version, hopefully.

### CODE REVIEW

**Comment by EamonNerbonne** [comment]
*ping*

**Comment by chrarnoldus** [comment]
Why do we need this? Can't we use `on delete cascade`?

**Comment by EamonNerbonne** [comment]
Because that doesn't work generally.

**Comment by EamonNerbonne** [comment]
Also, that requires a schema change, and thus allows accidental deletes possibly on otherwise harmless-looking queries, whereas this is more explicitly dangerous.

(I do believe we could use on delete cascade more frequently, however).

File: test/ProgressOnderwijsUtils.Tests/Data/CascadedDeleteTest.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            ").ExecuteNonQuery(Context);
+
+            var initialTLeafKeys = SQL($"select Z from TLeaf").ReadPlain<int>(Context);
+
+            PAssert.That(() => initialTLeafKeys.SetEqual(new[] { 1, 2, 3, 4 }));
+
+            CascadedDelete.RecursivelyDelete(Context, SQL($"TRoot"), new[] { 1, 2 }, null);
+
+            var finalT2 = SQL($"select D from T2").ReadPlain<int>(Context);
+            PAssert.That(() =>finalT2.SetEqual(new[] { 5 }));
```
Comment (markijbema) [inline test/ProgressOnderwijsUtils.Tests/Data/CascadedDeleteTest.cs]: ctrl-e-c

**Comment by EamonNerbonne** [comment]
and.... *ping*

**Comment by EamonNerbonne** [comment]
I'm going to interpret this silence as "uhhh... no idea" and merge this.

**Pull request merged**